Chess/src/Bishop.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Bishop.h"
+
+// It moves and captures along diagonals 
+vector<pair<int, int>> Bishop::getBasicMoves()
+{
+	return Bishop::getValidMoves();
+}
+
+std::vector<std::pair<int, int>> Bishop::getValidMoves()
+{
+	vector<pair<int, int>> validMoves;
+	// up left
+	auto upLeftMoves = getMovesWithDirection(-1, -1);
+	std::copy(upLeftMoves.begin(), upLeftMoves.end(), std::back_inserter(validMoves));
+
+	// up right
+	auto upRightMoves = getMovesWithDirection(-1, 1);
+	std::copy(upRightMoves.begin(), upRightMoves.end(), std::back_inserter(validMoves));
+
+	// down left
+	auto downLeftMoves = getMovesWithDirection(1, -1);
+	std::copy(downLeftMoves.begin(), downLeftMoves.end(), std::back_inserter(validMoves));
+
+	// down right
+	auto downRightMoves = getMovesWithDirection(1, 1);
+	std::copy(downRightMoves.begin(), downRightMoves.end(), std::back_inserter(validMoves));
+
+	return validMoves;
+}
+// Same function as in the Rook class. Consider moving it to a global scope.
+vector<pair<int, int>> Bishop::getMovesWithDirection(int rowOffset, int colOffset)
+{
+	vector<pair<int, int>> moves;
+
+	int currRow = _position.first + rowOffset;
+	int currCol = _position.second + colOffset;
+	while (currRow >= 0 && currRow < 8 && currCol >= 0 && currCol < 8)
```

Comment:
Extract the constant values to named variables: Instead of using hardcoded values like 8 for the board size, consider using named variables or constants. This makes the code more readable and allows for easier modifications in the future.